tests/components/alarmdecoder/test_config_flow.py

```diff
@@ -0,0 +1,424 @@
+        OPTIONS_ZONES: {},
+    }
+
+
+async def test_options_zone_flow_validation(hass: HomeAssistant):
+    """Test input validation for zone options flow."""
+    zone_number = "2"
+    zone_settings = {CONF_ZONE_NAME: "Front Entry", CONF_ZONE_TYPE: DEVICE_CLASS_WINDOW}
+    entry = MockConfigEntry(domain=DOMAIN)
+    entry.add_to_hass(hass)
+
+    result = await hass.config_entries.options.async_init(entry.entry_id)
+
+    assert result["type"] == data_entry_flow.RESULT_TYPE_FORM
+    assert result["step_id"] == "init"
+
+    result = await hass.config_entries.options.async_configure(
+        result["flow_id"],
+        user_input={"edit_selection": "Zones"},
+    )
+
+    assert result["type"] == data_entry_flow.RESULT_TYPE_FORM
+    assert result["step_id"] == "zone_select"
+
+    # Zone Number must be int
+    result = await hass.config_entries.options.async_configure(
+        result["flow_id"],
+        user_input={CONF_ZONE_NUMBER: "asd"},
+    )
+
+    assert result["type"] == data_entry_flow.RESULT_TYPE_FORM
+    assert result["step_id"] == "zone_select"
+    assert result["errors"] == {CONF_ZONE_NUMBER: "int"}
+
+    result = await hass.config_entries.options.async_configure(
+        result["flow_id"],
+        user_input={CONF_ZONE_NUMBER: zone_number},
+    )
+
+    assert result["type"] == data_entry_flow.RESULT_TYPE_FORM
+    assert result["step_id"] == "zone_details"
+
+    # CONF_RELAY_ADDR & CONF_RELAY_CHAN are inclusive
+    result = await hass.config_entries.options.async_configure(
+        result["flow_id"],
+        user_input={**zone_settings, CONF_RELAY_ADDR: "1"},
+    )
+
+    assert result["type"] == data_entry_flow.RESULT_TYPE_FORM
+    assert result["step_id"] == "zone_details"
+    assert result["errors"] == {"base": "relay_inclusive"}
+
+    result = await hass.config_entries.options.async_configure(
+        result["flow_id"],
+        user_input={**zone_settings, CONF_RELAY_CHAN: "1"},
+    )
+
+    assert result["type"] == data_entry_flow.RESULT_TYPE_FORM
+    assert result["step_id"] == "zone_details"
+    assert result["errors"] == {"base": "relay_inclusive"}
+
+    # CONF_RELAY_ADDR, CONF_RELAY_CHAN must be int
+    result = await hass.config_entries.options.async_configure(
+        result["flow_id"],
+        user_input={**zone_settings, CONF_RELAY_ADDR: "abc", CONF_RELAY_CHAN: "abc"},
+    )
+
+    assert result["type"] == data_entry_flow.RESULT_TYPE_FORM
+    assert result["step_id"] == "zone_details"
+    assert result["errors"] == {
+        CONF_RELAY_ADDR: "int",
+        CONF_RELAY_CHAN: "int",
+    }
+
+    # CONF_ZONE_LOOP depends on CONF_ZONE_RFID
+    result = await hass.config_entries.options.async_configure(
+        result["flow_id"],
+        user_input={**zone_settings, CONF_ZONE_LOOP: "1"},
+    )
+
+    assert result["type"] == data_entry_flow.RESULT_TYPE_FORM
+    assert result["step_id"] == "zone_details"
+    assert result["errors"] == {CONF_ZONE_LOOP: "loop_rfid"}
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=6158285)